src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -376,7 +376,7 @@ private void logInterpolationWarnings(String stepName, @CheckForNull ArgumentsAc
         }
 
         List<String> scanResults = sensitiveVariables.stream()
-                .filter(e -> !(envVars.get(e).isEmpty()) && interpolatedStrings.stream().anyMatch(g -> g.contains(envVars.get(e))))
+                .filter(e -> !(envVars.get(e, "").isEmpty()) && interpolatedStrings.stream().anyMatch(g -> g.contains(envVars.get(e))))
```

Comment:
nit: looks like there were some extra parentheses that could be removed:

```suggestion
                .filter(e -> !envVars.get(e, "").isEmpty() && interpolatedStrings.stream().anyMatch(g -> g.contains(envVars.get(e))))
```